test/framework/finalizers_helpers.go

```diff
@@ -149,6 +168,18 @@ func assertFinalizersExist(ctx context.Context, proxy ClusterProxy, namespace st
 	}).WithTimeout(1 * time.Minute).WithPolling(2 * time.Second).Should(Succeed())
 }
 
+// concatenateFinalizerAssertions concatenates all finalizer assertions into one map.
+func concatenateFinalizerAssertions(finalizerAssertions ...map[string][]string) map[string][]string {
+	allFinalizerAssertions := make(map[string][]string, 0)
+	for i := range finalizerAssertions {
+		for kind, finalizers := range finalizerAssertions[i] {
+			allFinalizerAssertions[kind] = finalizers
```

Comment:
Maybe we should throw an error if we find duplicate assertions for the same kind

Reply:
True, I added aggregated errors for such cases.